src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -188,7 +200,8 @@ class BlockchainImpl(
         removeBlockNumberMapping(block.number)
       else blockNumberMappingStorage.emptyBatchUpdate
 
-    val newBestBlockNumber: BigInt = (bestBlockNumber - 1).max(0)
+    val potientialNewBestBlockNumber: BigInt = (block.number - 1).max(0)
+    val potientialNewBestBlockHash: ByteString = block.header.parentHash
```

Comment:
There's a typo in `potential` in both variables! Should be `potentialNewBestBlockNumber` and `potentialNewBestBlockHash `

Reply:
I don't think so. It was already `bestBlockNumber` before and not `NewBestBlockNumber`.

I think my changes here deserve some explanations: 
By reading the code and how it is called, I saw that it actually assumes that the block we are deleting is the best block because it said that `newBestBlockNumber` is now `bestBlockNumber - 1` (so `bestBlockNumber == block.number`). I tried to make it clearer and less error prone by saying that the new best block number is actually `block.number - 1`.

Fixed the typo

Reply:
It's the fact that `bestBlockNumber` and `potentialNewBestBlockNumber` don't derive from each other anymore, makes me a bit uneasy. I think you ended up by adding two versions of getting the current best block

Reply:
In fact this part of the function will be removed after we merge #1092 so I think we should wait for #1092, fix the conflicts in this PR and we will be OK.

Reply:
👍

Reply:
#1092 is merged!